### DISCUSSION TITLE
Windows Terminal v1.15.2874 and v1.15.2875

### DISCUSSION THREAD

**Discussion by DHowett** [microsoft/terminal#14250] [Announcements]
v1.15.2874 (Windows 10) and v1.15.2875 (Windows 11) are servicing updates to Windows Terminal Stable v1.15. It is highly recommended that you install this version if you're using Windows 11 22H2 (October Update).

As a reminder, Terminal 1.12 was the last version of Windows Terminal that supports Windows 19H1 or 19H2.
Those versions of Windows went out of support in May, so you really may want to consider upgrading.

<details>
<summary>Preinstallation Kit info</summary>

A [preinstallation](https://docs.microsoft.com/en-us/windows/msix/desktop/deploy-preinstalled-apps) kit is available for system integrators and OEMs interested in prepackaging Windows Terminal with a Windows image. More information is available in the [DISM documentation on preinstallation](https://docs.microsoft.com/windows-hardware/manufacture/desktop/preinstall-apps-using-dism). Users who do not intend to preinstall Windows Terminal should continue using the _msixbundle_ distribution.

</details>

<details>
<summary>Why are there so many packages? How do I choose?</summary>
This version of Windows Terminal is distributed in two bundles, one of which works on Windows 10-11 and the other of
which only works on Windows 11. The Windows 11 version is much smaller because we no longer need to work around a
platform issue related to our dependencies.

If you intend on using Terminal as an unpackaged application--that is, extracting the `msix` file--we recommend that
you use the `Win10` bundle. You will need the [Visual C++ runtime redistributable](https://docs.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist).

In addition, if you install the _packaged_ version on either Windows 10 or Windows 11, it now depends on [the Visual C++ Universal Runtime Package](https://docs.microsoft.com/en-us/troubleshoot/developer/visualstudio/cpp/libraries/c-runtime-packages-desktop-bridge?msclkid=3e1dd82aac8b11ec8f9dbd86d5979ca8#how-to-install-and-update-desktop-framework-packages).

Despite these distributions having different version numbers, they are built from the same code and there is no
functional difference between them.

If you install the Windows 10 version on Windows 11, it will probably automatically upgrade itself to the Windows 11
version.
</details>

This servicing release supersedes the [v1.15.2712.0](https://github.com/microsoft/terminal/releases/tag/v1.15.2712.0) release (not uploaded to GitHub) and contains the following bug fixes and changes:

# Changes

* Terminal will now detect the title of a LNK or EXE as the default terminal (backport from 1.16) (#13570)

# Bug Fixes

## Stability and Security

* Windows Terminal once again works on Windows N (no media) SKUs
* We've fixed the following issues that could lead to a local denial of service or worse (with many thanks to @dgl for reporting):
   * `OSC 9;9` will now reject malformed paths and not further propagate them into `wsl --cd` (#14093) **REMOTE CODE EXECUTION** CVE-2022-44702
   * `OSC 8` URIs will be limited to 2MB in length, following iTerm2 (and only 1024 bytes will be displayed in the tooltip) (#14198)
* We've fixed some lag and deadlocking that would happen when you close a tab (#14041)
* We will no longer leak `OpenConsole` processes when they're running a Visual Studio Developer Shell (#14160)
* An application calling `system()` on its main thread will no longer deadlock itself and Terminal (when Terminal is set as default) (#14195)
* We've fixed a potential race condition causing a crash on tab close (#13882)

## Accessibility

* The WPF control now supports accessibility notifications (#14097)
* The Settings UI title/breadcrumb bar is now readable by screen readers (#14180)

## Usability

* You can now duplicate unfocused tabs (#13964) (thanks @JerBast!)
* The Open in Terminal (sic: Teeeerminal) shell extension should appear more reliably on the Desktop context menu (#14048) (#14211)
* "Export Text" will no longer suggest illegal filenames (#13693) (thanks @EliaSchiavon!)
* <kbd>Alt+Space</kbd> will now open the system menu in the Settings UI and Command Palette (#14221)
* `intenseTextStyle` is now included properly in the JSON schema (whoops) (#14210) (thanks @neersighted!)

---

_This discussion was created from the release [Windows Terminal v1.15.2874](https://github.com/microsoft/terminal/releases/tag/v1.15.2874.0)._

**Comment by neersighted** [comment]
It looks like an extra e snuck in ("Teerminal") @DHowett.

**Reply by DHowett** [reply]
Whoops, thanks!

**Reply by j4james** [reply]
@DHowett Did you just fix that "Teerminal" typo by adding even more e's? 😂

**Reply by DHowett** [reply]
I sure deeedn't!

**Reply by LucaBlackDragon** [reply]
Also, 7s and 8s are swapped in the version numbers inside the description (i.e. 1.15.2***78***4 instead of 1.15.2***87***4)

**Comment by Rayhubi** [comment]
👌

**Comment by JensGJ** [comment]
What's the reason behind the different version numbers for Windows 10 and Windows 11 (2874 vs. 2875)?

It is causing errors when I try to upgrade using Chocolatey (on my Windows 11 installation): 

```powershell
choco upgrade microsoft-windows-terminal
Chocolatey v1.2.0
Upgrading the following packages:
microsoft-windows-terminal
By upgrading, you accept licenses for the packages.

You have microsoft-windows-terminal v1.15.2524.0 installed. Version 1.15.2874.0 is available based on your source(s).
Progress: Downloading microsoft-windows-terminal 1.15.2874.0... 100%

microsoft-windows-terminal v1.15.2874.0 [Approved]
microsoft-windows-terminal package files upgrade completed. Performing other installation steps.
The package microsoft-windows-terminal wants to run 'chocolateyInstall.ps1'.
Note: If you don't run this script, the installation will fail.
Note: To confirm automatically next time, use '-y' or consider:
choco feature enable -n allowGlobalConfirmation
Do you want to run the script?([Y]es/[A]ll - yes to all/[N]o/[P]rint): A

WARNING: The installed 1.15.2874.0 version is newer than this package version, it may have autoupdated on your current OS...
ERROR: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Deployment failed with HRESULT: 0x80073D06, The package could not be installed because a higher version of this package is already installed.

Windows cannot install package Microsoft.WindowsTerminal_1.15.2874.0_x64__8wekyb3d8bbwe because it has version 1.15.2874.0. A higher version 1.15.2875.0 of this package is already installed.
```

So somehow the installed v1.15.2875.0 version identifies itself as v1.15.2524.0 to the chocolatey package system - until the actual installation process finds out that it is indeed  v1.15.2875.0  that is installed.

**Reply by DHowett** [reply]
Ah, interesting. I think there's two things going on there.

1. Chocolatey's database is out of sync with the installed AppX package catalog. The last time you got an update from Chocolatey, it installed `1.15.2524`, and Chocolatey recorded that for its own purposes. Later, you got an automatic store update to `1.15.2875`. _This seems to be a bug in Chocolatey's Windows Store integration._
2. The package available in Chocolatey's repository is the "Windows 10" version of the package. You already know this 😄 

So, why do we have two different versions?

There's an OS bug in Win10 that prevents us from running as Administrator. To fix it, we need to include some extra DLLs in our package. We didn't want to bloat the package for everybody, though, so we split it into two versions: one for broken versions of Windows, and one for fixed versions.

However, the Store can't handle two copies of the same _version number_ being uploaded . . . so we had to make them appear as different versions. Ugh. FWIW, I hate this hack.

Anyway, for best compatibility and since Chocolatey and Winget lack OS-version-based targeting, they only include the Win10 version of our package. It's compatible with both 10 and 11.

I'm not sure how to poke Chocolatey into thinking that you have the most up-to-date version so it stops trying and failing to install it.

**Comment by r-rajaneesh** [comment]
How do I exit VIM?

**Reply by zadjii-msft** [reply]
[How do I exit VIM?](https://lmgt.org/?q=How+do+I+exit+VIM%3F)

**Comment by DHowett** [comment]
Release notes updated to indicate that PR #14093 was targeted to address a remote code execution vulnerability assigned ID **CVE-2022-44702** (`AV:L/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H/E:U/RL:O/RC:C`)

Thanks again to @dgl for the report!
